src/cmds/pbs_rstat.c

```diff
@@ -98,7 +99,7 @@ display_single_reservation(struct batch_status *resv, int how)
 				else if (strcmp(attrp->name, ATTR_resv_end) == 0)
 					resv_end = attrp->value;
 				else if (strcmp(attrp->name, ATTR_resv_duration) == 0)
-					resv_duration = atoi(attrp->value);
+					resv_duration = atol(attrp->value);
```

Comment:
Please use strtoll() instead. ato* functions cannot identify between invalid input in all cases. It would be better if you strtoll() and make sure errno is not set.

Reply:
Instead of strtoll(), used strtol(). As the datatype of resv_duration() is long int in the code.